shared/src/main/java/com/faforever/neroxis/mask/FloatMask.java

```diff
@@ -171,7 +181,43 @@ public FloatMask addPerlinNoise(int resolution, float scale) {
             float noiseMin = noise.getMin();
             float noiseMax = noise.getMax();
             float noiseRange = noiseMax - noiseMin;
-            noise.apply((x, y) -> noise.setPrimitive(x, y, (noise.getPrimitive(x, y) - noiseMin) / noiseRange * scale));
+            noise.apply((x, y) -> {
+                noise.setPrimitive(x, y, (noise.getPrimitive(x, y) - noiseMin) / noiseRange * scale);
+            });
+
+            // Some pixels have not been set when there is Odd symmetry, lets find them
+            // and fix them by taking an average of the surrounding pixels.
+            if (symmetrySettings.getSymmetry(SymmetryType.SPAWN).isOddSymmetry()) {
+                for (int i = 0; i < size; i++) {
+                    for (int j = 0; j < size; j++) {
+                        if (noise.hasSet[i][j] == false) {
+                            int numAround = 0;
+                            float total = 0;
+                            if (i > 0 && noise.hasSet[i - 1][j]) {
+                                total += noise.mask[i - 1][j];
+                                numAround++;
+                            }
+                            if (j > 0 && noise.hasSet[i][j - 1]) {
+                                total += noise.mask[i][j - 1];
+                                numAround++;
+                            }
+                            if (i < size - 1 && noise.hasSet[i + 1][j]) {
+                                total += noise.mask[i + 1][j];
+                                numAround++;
+                            }
+                            if (j < size - 1 && noise.hasSet[i][j + 1]) {
+                                total += noise.mask[i][j + 1];
+                                numAround++;
+                            }
+                            if (numAround > 0) {
+                                noise.mask[i][j] = total / numAround;
+                            }
+                        }
+                    }
+                }
+                // Free up the memory
+                noise.hasSet = null;
```

Comment:
Unfortunately this doesn't free up the memory right away so can result in heavier load on the users computer